### PR TITLE
Jetpack: Update network and db calls to adding various missing site settings

### PR DESCRIPTION
**Pull request by tonyr59h**
This PR is #6741's sibling. It updates network serialization methods for fetching and updating the new options. It also adds columns to the database and handles migration.

To test:
* Since there's no way to see these changes in the app I'd just set a breakpoint [here](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/6652-missing-settings-master...issue/6652-missing-settings-network#diff-c7e7383fc8aa13adeb5e7f0a75d559b2R208) and make sure all the values are set correctly.

### CODE REVIEW

**Comment by kwonye** [comment]
@tonyr59h can we have this PR go into #6741 and then #6741 into the feature branch? That way the dependency is 100% clear, and the UI/db calls all go in as one PR.

File: WordPress/src/main/java/org/wordpress/android/models/SiteSettingsModel.java [hunk 1]
```diff
@@ -288,6 +323,13 @@ public void deserializeOptionsDatabaseCursor(Cursor cursor, Map<Integer, Categor
         commentsRequireIdentity = getBooleanFromCursor(cursor, IDENTITY_REQUIRED_COLUMN_NAME);
         commentsRequireUserAccount = getBooleanFromCursor(cursor, USER_ACCOUNT_REQUIRED_COLUMN_NAME);
         commentAutoApprovalKnownUsers = getBooleanFromCursor(cursor, WHITELIST_COLUMN_NAME);
+        timezone = getStringFromCursor(cursor, "siteTimezone");
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/models/SiteSettingsModel.java]: Please make these constants

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java [hunk 2]
```diff
@@ -204,6 +205,13 @@ public void deserializeDotComRestResponse(SiteModel site, JSONObject response) {
         mRemoteSettings.sharingButtonStyle = settingsObject.optString(SHARING_BUTTON_STYLE_KEY, DEFAULT_SHARING_BUTTON_STYLE);
         mRemoteSettings.allowCommentLikes = settingsObject.optBoolean(SHARING_COMMENT_LIKES_KEY, false);
         mRemoteSettings.twitterUsername = settingsObject.optString(TWITTER_USERNAME_KEY, "");
+        mRemoteSettings.timezone = settingsObject.optString("timezone_string", "");
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: Please make these string constants

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java [hunk 3]
```diff
@@ -340,26 +348,42 @@ public void deserializeDotComRestResponse(SiteModel site, JSONObject response) {
...
         }
-
         if (mSettings.allowCommentLikes != mRemoteSettings.allowCommentLikes) {
             params.put(SHARING_COMMENT_LIKES_KEY, String.valueOf(mSettings.allowCommentLikes));
         }
-
         if (mSettings.twitterUsername != null && !mSettings.twitterUsername.equals(mRemoteSettings.twitterUsername)) {
             params.put(TWITTER_USERNAME_KEY, mSettings.twitterUsername);
         }
+        if (!StringUtils.equals(mSettings.timezone, mRemoteSettings.timezone)) {
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: Same

**Comment by tonyr59h** [comment]
> can we have this PR go into #6741

Re-based it for ya 👍 

Thanks for the review @kwonye!

**Comment by kwonye** [comment]
I'm running into this error when testing the upgrade path from `develop`

```
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: org.wordpress.android.beta, PID: 27698
                  java.lang.RuntimeException: Unable to create application org.wordpress.android.WordPressDebug: android.database.sqlite.SQLiteException: Can't downgrade database from version 16 to 15. Consider implementing method getOnDowngradeListener of interface WellConfig
                      at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5794)
                      at android.app.ActivityThread.-wrap1(Unknown Source:0)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
                      at android.os.Handler.dispatchMessage(Handler.java:105)
                      at android.os.Looper.loop(Looper.java:164)
                      at android.app.ActivityThread.main(ActivityThread.java:6541)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                   Caused by: android.database.sqlite.SQLiteException: Can't downgrade database from version 16 to 15. Consider implementing method getOnDowngradeListener of interface WellConfig
                      at com.yarolegovich.wellsql.DefaultWellConfig.onDowngrade(DefaultWellConfig.java:89)
                      at com.yarolegovich.wellsql.WellSql.onDowngrade(WellSql.java:52)
                      at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:297)
                      at android.database.sqlite.SQLiteOpenHelper.getReadableDatabase(SQLiteOpenHelper.java:218)
                      at com.yarolegovich.wellsql.WellSql.select(WellSql.java:69)
                      at org.wordpress.android.fluxc.persistence.AccountSqlUtils.getAccountByLocalId(AccountSqlUtils.java:80)
                      at org.wordpress.android.fluxc.persistence.AccountSqlUtils.getDefaultAccount(AccountSqlUtils.java:71)
                      at org.wordpress.android.fluxc.store.AccountStore.loadAccount(AccountStore.java:682)
                      at org.wordpress.android.fluxc.store.AccountStore.<init>(AccountStore.java:395)
                      at org.wordpress.android.fluxc.store.AccountStore_Factory.get(AccountStore_Factory.java:39)
                      at org.wordpress.android.fluxc.store.AccountStore_Factory.get(AccountStore_Factory.java:13)
                      at dagger.internal.ScopedProvider.get(ScopedProvider.java:46)
                      at org.wordpress.android.WordPress_MembersInjector.injectMembers(WordPress_MembersInjector.java:52)
                      at org.wordpress.android.WordPress_MembersInjector.injectMembers(WordPress_MembersInjector.java:15)
                      at org.wordpress.android.modules.DaggerAppComponent.inject(DaggerAppComponent.java:579)
                      at org.wordpress.android.WordPress.onCreate(WordPress.java:208)
                      at org.wordpress.android.WordPressDebug.onCreate(WordPressDebug.java:13)
                      at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
                      at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
                      at android.app.ActivityThread.-wrap1(Unknown Source:0) 
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661) 
                      at android.os.Handler.dispatchMessage(Handler.java:105) 
                      at android.os.Looper.loop(Looper.java:164) 
                      at android.app.ActivityThread.main(ActivityThread.java:6541) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
```

**Pull request merged**